bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/utils/Utils.java

```diff
@@ -12,17 +12,52 @@
  */
 package org.openhab.binding.nikobus.internal.utils;
 
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
 import java.util.concurrent.Future;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
 /**
  * The {@link Utils} class defines commonly used utility functions.
  *
  * @author Boris Krivonog - Initial contribution
  */
+@NonNullByDefault
 public class Utils {
     public static void cancel(Future<?> future) {
```

Comment:
The error you are seeing in the build are mostly referring to this method. Since you added `@NonNullByDefault` to this class it will assume that all fields and parameters are non-null.
This `cancel` method has now been changed to enforce that the `future` parameter is non-null, so any callers of this method must also make sure that they are passing non-null futures. 
It is apparent from the implementation of this method that the `future` parameter was intended to be nullable, as were all the callers of this method assumed it could accept null as well.
In short, to fix the errors just make the parameter nullable.
```suggestion
    public static void cancel(@Nullable Future<?> future) {
```